### ISSUE TITLE
Compose flag `depends_on` gets ignored

### ISSUE THREAD

**Issue by davidkopp** [green-coding-berlin/green-metrics-tool#567]
GMT currently doesn't handle service dependencies defined by the Docker Compose key [depends_on](https://docs.docker.com/compose/compose-file/05-services/#depends_on). That is a problem, because in usage scenarios there could be dependencies between services that have to be respected during startup. If not, the startup of services could fail.

Known workaround:
Change the order of services in the compose file.

**Comment by ArneTR** [comment]
Thanks for bringing this up in a new issue.

This is also relevant / partial duplicate of: https://github.com/green-coding-berlin/green-metrics-tool/issues/423

Now that we have another issue mentioning this it should maybe be prioritized.

@davidkopp : Could you have a look at the referenced #423 and add you statement here, or in the other issue, how you use `depends_on` and if you just need container boot guarantees or actually have checks running. If the latter: What are they?

For us it is important to understand to which extent that feature is used in the wild

**Comment by davidkopp** [comment]
For me `depends_on` is mainly relevant to ensure a correct startup order of containers as the Compose specification states:
> Compose guarantees dependency services have been started before starting a dependent service.

I don't want that my web backend starts before my database is started (and is ready).

An additional health check of the containers during startup would definitely help to determine the best startup time, so no container gets started to early and to avoid unnecessary waiting times if GMT uses a predefined sleep time as of now.

So I think it makes sense to have two different issues:
1. ensure correct startup order (#567)
2. implement health checks (#423)

I'm currently working on implementing the first part.
